bundles-lib/src/main/java/org/apache/metron/bundles/BundleThreadContextClassLoader.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.bundles;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.metron.bundles.bundle.Bundle;
+import org.apache.metron.bundles.util.BundleProperties;
+
+/**
+ * THREAD SAFE
+ */
+public class BundleThreadContextClassLoader extends URLClassLoader {
+
+    static final ContextSecurityManager contextSecurityManager = new ContextSecurityManager();
+    private final ClassLoader forward = ClassLoader.getSystemClassLoader();
+    private static final List<Class<?>> bundleSpecificClasses = new ArrayList<>();
+    private static AtomicBoolean inited = new AtomicBoolean(false);
+    // should initialize class definitions
+    public static void initClasses(final List<Class> classes){
```

Comment:
The comparable code in Nifi used a static initializer to init the classes list.
This uses a public method, and an AtomicBoolean flag to say it's been done.
Why doesn't the initClasses() method need to be synchronized?

(BTW, I really dislike these all-static or mostly-static classes, which seem to be trying to be singletons without engaging in the discipline of singleton-ness, especially wrt thread safeness.  But I understand they originated in the nifi code, so I'm not after you about that :-)